PayloadPkg/FirmwareUpdate/CsmeFwUpdate.c

```diff
@@ -76,6 +77,76 @@ DisplaySendStatus (
   IN UINT32 totalBytesToSendToFw
   );
 
+/**
+  Reads a range of PCI configuration registers into a caller supplied buffer.
+
+  Reads the range of PCI configuration registers specified by StartAddress and
+  Size into the buffer specified by Buffer. This function only allows the PCI
+  configuration registers from a single PCI function to be read. Size is
+  returned. When possible 32-bit PCI configuration read cycles are used to read
+  from StartAddress to StartAddress + Size. Due to alignment restrictions, 8-bit
+  and 16-bit PCI configuration read cycles may be used at the beginning and the
+  end of the range.
+
+  If StartAddress > 0x0FFFFFFF, then ASSERT().
+  If ((StartAddress & 0xFFF) + Size) > 0x1000, then ASSERT().
+  If Size > 0 and Buffer is NULL, then ASSERT().
+
+  @param  StartAddress  The starting address that encodes the PCI Bus, Device,
+                        Function and Register.
+  @param  Size          The size in bytes of the transfer.
+  @param  Buffer        The pointer to a buffer receiving the data read.
+
+  @return EFI_SUCCESS        if data is read into buffer
+  @return EFI_NOT_FOUND      if data is NOT read into buffer
+  @return EFI_INVALID_PARAMETER  Invalid parameter
+**/
+EFI_STATUS
+EFIAPI
+CsmePciReadBuffer (
+  IN      UINTN     StartAddress,

```

Comment:
who will provide this StartAddress and who will directly call this function?
we have a Macro TO_PCI_LIB_ADDRESS (), not sure we could use that one.

Reply:
I looked at the Macro, that does not work.

CSME firmware update library will call this function and will pass StartAddress.